### PR TITLE
Event creator

### PR DESCRIPTION
**Pull request by maykremer**

### CODE REVIEW

File: CreateEvent/readme.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Event Queue Triggeer
```
Comment (carmeltsadok) [inline CreateEvent/readme.md:1]: Not sure it is needed. There are a lot of outdated interfaces

File: shared/event/event.model.ts [hunk 2]
```diff
@@ -11,15 +11,14 @@ const eventSchema = new Schema(
...
-                required: true,
             },
             name: {
                 type: String,
                 required: true,
             },
         },
         malshabShort: {
-            identityNumber: {
+            id: {
```
Comment (carmeltsadok) [inline shared/event/event.model.ts:21]: needs to check with vitaly

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: CreateEvent/index.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        await getConnection();
+
+        if (typeof myQueueItem === 'string') {
+            throw new Error('The create event message arrived as string instead of a JSON object, the object is probably invalid.');
+        }
+
+        const queueObj = myQueueItem as Event;
+
+        const { error } = validEvent.validate(queueObj);
+        if (error) throw new Error(error.message);
```
[the remark this answers is not in the document]
Comment (carmeltsadok) [reply CreateEvent/index.ts:19]: Change to 
`const validationError: ValidationError = getReportsSchema.validate(req).error;        if (validationError) {            throw new ValidationFunctionError(functionName, validationError.message);        }`

**Review by carmeltsadok — APPROVED** [review]

File: CreateEvent/index.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const functionName = 'createEvent';
+
+const createEvent: AzureFunction = async (context: Context, myQueueItem: any): Promise<void> => {
+    try {
+        await getConnection();
+
+        if (typeof myQueueItem === 'string') {
+            throw new Error('The create event message arrived as string instead of a JSON object, the object is probably invalid.');
+        }
+
```
Comment (lustigy96) [inline CreateEvent/index.ts:19]: I think that if we suppose a good input, so this check is unneeded.. 
and if we don't, so it's better to check that myQueueItem is a JSON object, and not just ensure that it is not a string.

**Review by lustigy96 — CHANGES_REQUESTED** [review]

**Review by lustigy96 — APPROVED** [review]

**Pull request merged**